jormungandr/src/explorer.rs

```diff
@@ -0,0 +1,86 @@
+use super::blockchain::Blockchain;
+use crate::blockcfg::Block;
+use crate::blockcfg::ChainLength;
+use crate::intercom::ExplorerMsg;
+use crate::utils::task::{Input, ThreadServiceInfo};
+use chain_impl_mockchain::multiverse::{GCRoot, Multiverse};
+use futures::Future;
+use futures::IntoFuture;
+
+use chain_core::property::Block as _;
+
+pub struct Process {
+    multiverse: Multiverse<State>,
+}
+
+struct State {
+    block: Block,
+}
+
+impl State {
+    pub fn new(block: Block) -> Self {
+        Self { block }
+    }
+}
+
+impl Process {
+    pub fn new() -> Self {
+        Self {
+            multiverse: Multiverse::<State>::new(),
+        }
+    }
+
+    pub fn handle_input(
+        &mut self,
+        info: &ThreadServiceInfo,
+        blockchain: &Blockchain,
+        input: Input<ExplorerMsg>,
+    ) {
+        let logger = info.logger();
+        let bquery = match input {
+            Input::Shutdown => {
+                return;
+            }
+            Input::Input(msg) => msg,
+        };
+
+        match bquery {
+            ExplorerMsg::NewBlock(header) => {
+                //I don't really understand what's the purpose of the gcroot yet
```

Comment:
so the multiverse is a way to handle multiple/possible forks. It allows you to follow different branches where these branches might still share some common data (here the same parent block for example). This cannot grow indefinitely, so there is some sort of garbage collection available on demand (this is not used at the moment, we are kind of trying to see how far we can go in term of memory waster :D in the node).

Now if you drop the `GCRoot`, if someone calls Multiverse's function `gc` https://github.com/input-output-hk/chain-libs/blob/09b67a1d7117e103edc7f8b62b5a659af87692dc/chain-impl-mockchain/src/multiverse.rs#L159 then the code `Block` will be removed from the `Multiverse`.

Reply:
Alright, I understand now. Although from what I see, `gc` is only implemented for the Ledger specialization.